### ISSUE TITLE
isIdResponse() doesn't catch some non-compliant hardware

### ISSUE THREAD

**Issue by jeffgca** [jazz-soft/JZZ-midi-Gear#8]
isIdResponse correctly tests that the 5th byte has a value of 2:

https://github.com/jazz-soft/JZZ-midi-Gear/blob/master/javascript/JZZ.midi.Gear.js#L15

```javascript
JZZ.MIDI.prototype.isIdResponse = function() {
  return this[0] == 0xf0 && this[1] == 0x7e && this[3] == 6 && this[4] == 2 && this[this.length-1] == 0xf7 && this.length > 10;
};
```

Unfortunately some synth manufacturers do not send a correct device inquiry response. I happen to own a Waldorf Microwave XT, and the response it sends is:

```
F0 7E 06 02 3E 0E 00 0B  00 32 2E 33 33 F7
```

This is documented thoroughly [in the manual](https://www.waldorfmusic.com/en/hardware-archive/microwave-iixtxtk-series/microwave-ii-xt-xtk-series-downloads/category/210-Documentation):

```
The Microwave 2 will respond with the following:

F0,7E,06,02        Universal Device Header
```

`02` is the 4th byte! 🤬

The instrument has the last available firmware installed ( 2.33 ) so this is unlikely to ever get fixed by Waldorf, and regardless has now likely been baked into third party software. I'm not sure what the right approach here is, it should be reliable to test for a value of 2 at the 5th byte but I thought I'd log the issue so at least other MWXT owners don't go crazy. 

An alternative approach I might adopt for some other work I'm doing is to key all known instruments using this utility function:

```
msg = [240, 126, 6, 2, 62, 14, 0, 11, 0, 50, 46, 51, 51, 247];
function makeUniqueId(intArr) {
    return intArr.map((int) => { return int.toString(16).padStart(2, '0'); }) 
}
console.log(makeUniqueId(msg))
// returns 'f0-7e-06-02-3e-0e-00-0b-00-32-2e-33-33-f7'
```

It's unlikely there would be collisions and I can then support non-standard hardware without special logic and I'd have O(1) lookups.

**Comment by jazz-soft** [comment]
Good question...
Definitely the response does not follow the MIDI specs, it skips the 3rd byte.
Need to think what to do about that...

How do the other Waldorf devices behave?

**Comment by jeffgca** [comment]
My Blofeld responds correctly so they started doing this correctly at some point - unsure about other models as Waldorf doesn't seem to publish sysex documentation for recent models - the Blofeld for example only has controller message docs in the manual, and devices like the iridium / quantum don't have anything like this.

**Comment by jazz-soft** [comment]
Please take a look at my latest commit before I publish a new release at npm.

**Comment by jeffgca** [comment]
I think the correct sysex response from the MWXT ( mine at least ) is this:

`f0 7e 06 02 3e 0e 00 0b 00 32 2e 33 33 f7	 Waldorf	Microwave XT	Synth Module`

Note: missing `7f` byte at index 2, length should be 14 instead of 15

**Comment by jazz-soft** [comment]
Yes, I have added an extra byte in the table for the data compiler, but it will understand the incomplete message from your device.
See the other change in the script.

**Comment by jazz-soft** [comment]
Published v1.1.9 at npm
